views/application.py

```diff
@@ -93,3 +97,37 @@ def update(self, request, *args, **kwargs):
                 list(set(new_team_member_ids)-set(old_team_member_ids))
             )
         return response.Response(serializer.data)
+
+    
+class ApplicationHiddenDetailView(generics.GenericAPIView):
+    """
+    View details of Application which should not be exposed directly 
+    """
+
+    permission_classes = [permissions.IsAuthenticated, ]
+
+    def post(self,request,*args,**kwargs):
+
+        """
+        Authenticate if the requesting user is Team Member
+        of the application and check the password entered.
+        :param request: the request being processed
+        :param args: arguments
+        :param kwargs: keyword arguments
+        :return:response containing client_secret
+        """
+
+        try:
+            application = Application.objects.get(pk = request.data.get('id'))
+            application_team_members = application.team_members.all()
+            for team_member in application_team_members:
```

Comment:
You may use Django ORM `filter` and exist command to check if the person is a team member. This for loop is unnecessary.

Reply:
Yes sure.
Can we use `.get` in place of `filter` so that we can use the object directly to check_password ?
```
a = Application.objects.get(pk = request.data.get('id'))
p = request.person
a.team_members.get(id = p.id).user.check_password("requested_password") 
```

Reply:
This will not handle the edge cases, get throws a 500 if the object doesn't exist. So, yes you can use .get but you need to take care of handling the various exceptions that can occur and maintain the code quality. Fo example, we generally don't use so much nested `.` together as each property can raise an error. So we extract things one by one handling cases.

Reply:
Yes right.
Thank you for the explanation.